src/spec-lang/tc/typecheck.ts

```diff
@@ -84,7 +87,12 @@ export class StateVarScope {
         public readonly target: VariableDeclaration,
```

Comment:
If we add an `implements PPAble` to the class definition for `StateVarScope` can we avoid the casts to `scope as PPAble` further down in this file?

Reply:
Good catch. Addressed in https://github.com/ConsenSys/scribble/pull/94/commits/d4ed33eadbc26a96e9b7719cff33f967783af572.